Jenkinsfile

```diff
@@ -322,27 +322,22 @@ pipeline {
                 }
             }
         }
-        stage('Get Commit Message') {
-            steps {
-                script {
-                    env.COMMIT_MESSAGE = sh(script: 'git show -s --format=%B',
-                                            returnStdout: true).trim()
-                    Map pragmas = [:]
-                    // can't use eachLine() here: https://issues.jenkins.io/browse/JENKINS-46988/
-                    env.COMMIT_MESSAGE.split('\n').each { line ->
-                        String key, value
-                        try {
-                            (key, value) = line.split(':', 2)
-                            if (key.contains(' ')) {
-                                return
-                            }
-                            pragmas[key.toLowerCase()] = value
-                        /* groovylint-disable-next-line CatchArrayIndexOutOfBoundsException */
-                        } catch (ArrayIndexOutOfBoundsException ignored) {
-                            // ignore and move on to the next line
+        stage('Prepare Environment Variables') {
+            parallel {
+                stage('Get Commit Message') {
+                    steps {
+                        pragmasToEnv()
+                    }
+                }
+                stage('Determine Base Branch') {
+                    steps {
+                        script {
+                            env.BASE_BRANCH_NAME = sh(label: 'Determine base branch name',
+                                                      script: 'utils/rpms/packaging/get_base_branch',
+                                                      returnStdout: true).trim()
+                            echo 'Base branch == ' + env.BASE_BRANCH_NAME
```

Comment:
Ha.  I had forgotten about env.CHANGE_TARGET.  Although the argument that could be made for `get_base_branch` is that it's independent of infrastructure like GitHub or GitLab, etc and works with `git` alone.  I'm not sure that's an important consideration at this point though.

Reply:
Actually, I take that back, and am going to have to update this PR to reflect that.  `env.CHANGE_TARGET` is not useful in this case as it could be any arbitrary branch.  Any arbitrary branch doesn't tell us the ultimate landing branch which is what we are looking for here.  Using the _base branch_ nomenclature in this stage is probably misleading since GH uses that nomenclature to describe what `env.CHANGE_BRANCH` is -- the merge target of a change -- but that's not what we are looking for technically.